craft_platforms/_distro.py

```diff
@@ -0,0 +1,140 @@
+    """Get the version of a base."""
+    if isinstance(base, BaseName):
+        return base.version
+    return base[1]
+
+
+@dataclasses.dataclass(repr=True)
+class DistroBase:
+    """A linux distribution base."""
+
+    name: str
+    version: str
+
+    def _ensure_bases_comparable(self, other: BaseName | tuple[str, str]) -> None:
+        """Ensure that these bases are comparable, raising an exception if not.
+
+        :param other: Another distribution base.
+        :raises: ValueError if the distribution bases are not comparable.
+        """
+        other_name = other.name if isinstance(other, BaseName) else other[0]
+        if self.name != other_name:
+            raise ValueError(
+                f"Different distributions ({self.name} and {other_name}) do not have comparable versions.",
+            )
+
+    def __eq__(self, other: object, /) -> bool | NotImplementedType:
+        if isinstance(other, BaseName):
+            return self.name == other.name and self.version == other.version
+        if isinstance(other, tuple):
+            return bool(self.name == other[0] and self.version == other[1])
+        return NotImplemented
+
+    def __lt__(self, other: BaseName | tuple[str, str]) -> bool:
+        self._ensure_bases_comparable(other)
+        self_version_tuple = _get_version_tuple(self.version)
+        other_version_tuple = _get_version_tuple(_get_version(other))
+        return self_version_tuple < other_version_tuple
+
+    def __le__(self, other: BaseName | tuple[str, str]) -> bool:
+        self._ensure_bases_comparable(other)
+        self_version_tuple = _get_version_tuple(self.version)
+        other_version_tuple = _get_version_tuple(_get_version(other))
+        return self_version_tuple <= other_version_tuple
+
+    def __gt__(self, other: BaseName | tuple[str, str]) -> bool:
+        self._ensure_bases_comparable(other)
+        self_version_tuple = _get_version_tuple(self.version)
+        other_version_tuple = _get_version_tuple(_get_version(other))
+        return self_version_tuple > other_version_tuple
+
+    def __ge__(self, other: BaseName | tuple[str, str]) -> bool:
+        self._ensure_bases_comparable(other)
+        self_version_tuple = _get_version_tuple(self.version)
+        other_version_tuple = _get_version_tuple(_get_version(other))
+        return self_version_tuple >= other_version_tuple
+
+    @classmethod
+    def from_str(cls, base_str: str) -> Self:
+        """Parse a distribution string to a DistroBase.
+
+        :param base_str: A distribution string (e.g. "ubuntu@24.04")
+        :returns: A DistroBase of this string.
+        :raises: ValueError if the string isn't of the appropriate format.
+        """
+        if base_str.count("@") != 1:
+            raise ValueError(
+                f"Invalid base string {base_str!r}. Format should be 'distro@version'",
+            )
+        name, _, version = base_str.partition("@")
```

Comment:
Would this be a good opportunity to call this term "distro" or "name" but not both?